### PR TITLE
refactor: add base plugin and rename plugins to follow convention

### PR DESCRIPTION
**Pull request by mydiemho**
- [x]  All plugin classes now extends from base plugin
- [x] All plugin classes rename to follow naming convention

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ node_modules/
 /lib/
 *.dll
 /coverage/
-.env*
\ No newline at end of file
+.env*
+.idea/
```
Comment (wbreza) [inline .gitignore]: What is `.idea` ??

Comment (tbarlow12) [reply .gitignore]: that sounds like intellij stuff

Comment (tbarlow12) [reply .gitignore]: or probably webstorm I'm guessing

**Comment by tbarlow12** [comment]
How did coverage decrease 10%?

File: .gitignore (hunk 1 above)
Comment (mydiemho) [reply .gitignore]: yup intellij/webstorm metadata

**Comment by mydiemho** [comment]
> How did coverage decrease 10%?

looks like a few plugins didn't have enough tests, and renaming them cause them to be counted

**Pull request merged**